jbpm-task-assigning/kie-server-client-task-assigning/src/main/java/org/kie/server/client/impl/TaskAssigningRuntimeClientImpl.java

```diff
@@ -0,0 +1,144 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.client.impl;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kie.server.api.exception.KieServicesException;
+import org.kie.server.api.model.taskassigning.ExecutePlanningResult;
+import org.kie.server.api.model.taskassigning.LocalDateTimeValue;
+import org.kie.server.api.model.taskassigning.PlanningItemList;
+import org.kie.server.api.model.taskassigning.TaskDataList;
+import org.kie.server.api.model.taskassigning.TaskInputVariablesReadMode;
+import org.kie.server.api.rest.RestURI;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.TaskAssigningRuntimeClient;
+
+import static org.kie.server.api.model.taskassigning.QueryParamName.FROM_LAST_MODIFICATION_DATE;
+import static org.kie.server.api.model.taskassigning.QueryParamName.FROM_TASK_ID;
+import static org.kie.server.api.model.taskassigning.QueryParamName.PAGE;
+import static org.kie.server.api.model.taskassigning.QueryParamName.PAGE_SIZE;
+import static org.kie.server.api.model.taskassigning.QueryParamName.STATUS;
+import static org.kie.server.api.model.taskassigning.QueryParamName.TASK_INPUT_VARIABLES_MODE;
+import static org.kie.server.api.model.taskassigning.TaskAssigningRestURI.TASK_ASSIGNING_EXECUTE_PLANNING_URI;
+import static org.kie.server.api.model.taskassigning.TaskAssigningRestURI.TASK_ASSIGNING_QUERIES_TASK_DATA_URI;
+import static org.kie.server.api.model.taskassigning.TaskAssigningRestURI.TASK_ASSIGNING_RUNTIME_URI;
+
+public class TaskAssigningRuntimeClientImpl extends AbstractKieServicesClientImpl implements TaskAssigningRuntimeClient {
+
+    public TaskAssigningRuntimeClientImpl(KieServicesConfiguration config) {
+        super(config);
+    }
+
+    public TaskAssigningRuntimeClientImpl(KieServicesConfiguration config, ClassLoader classLoader) {
+        super(config, classLoader);
+    }
+
+    @Override
+    public ExecutePlanningResult executePlanning(PlanningItemList planningItemList, String userId) {
+        if (config.isRest()) {
+            final String uri = TASK_ASSIGNING_RUNTIME_URI + "/" + TASK_ASSIGNING_EXECUTE_PLANNING_URI + "?user=" + userId;
+            return makeHttpPostRequestAndCreateCustomResponse(RestURI.build(loadBalancer.getUrl(),
+                                                                            uri,
+                                                                            Collections.emptyMap()),
+                                                              planningItemList,
+                                                              ExecutePlanningResult.class);
+        } else {
+            throw new KieServicesException("JMS protocol is not implemented for this service.");
```

Comment:
Can we get away with this? Is it normal that some KIE server components would not implement JMS, only REST?

Reply:
at this point the only protocol that is being used is REST and so will probalby happen with 95% probability for the rest of the life of this API.  Note that is api is used for consuming the tasks information and executing the plan from the KieServerPlanningExtension, and here we don't want/need JMS invocation.
In general I'd say that this also the for 90% of the uses of the kie-server apis, and only a reduced set of use cases/customers use JMS.
So by now we can live with that, but yes, in future iteration JMS alternative can be added with no problem.